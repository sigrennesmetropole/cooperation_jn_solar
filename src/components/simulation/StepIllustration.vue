<script setup lang="ts">
defineProps({
  image: {
    type: String,
    required: true,
  },
  text: {
    type: String,
    required: true,
  },
  selected: {
    type: Boolean,
    default: false,
  },
})
</script>

<template>
  <div class="flex flex-col items-center py-2 px-3 grow">
    <img :src="image" alt="" :class="selected ? '' : 'opacity-30'" />
    <p
      class="font-dm-sans font-medium text-xs"
      :class="selected ? '' : 'text-neutral-400'"
      id="step-title"
    >
      {{ text }}
    </p>
  </div>
</template>
