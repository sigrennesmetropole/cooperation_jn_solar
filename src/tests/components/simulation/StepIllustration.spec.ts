import StepIllustration from '@/components/simulation/StepIllustration.vue'
import { describe, it, expect, beforeEach } from 'vitest'
import { mount, VueWrapper } from '@vue/test-utils'

describe('StepIllustration.vue', () => {
  let wrapper: VueWrapper

  beforeEach(async () => {
    wrapper = mount(StepIllustration, {
      propsData: {
        image: 'potentiel.svg',
        text: 'Ensoleillement',
        selected: true,
      },
    })
  })

  it('renders the component correctly', () => {
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
