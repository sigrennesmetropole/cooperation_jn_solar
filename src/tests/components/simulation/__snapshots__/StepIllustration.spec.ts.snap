// Vitest Snapshot v1

exports[`StepIllustration.vue > renders the component correctly 1`] = `
"<div class=\\"flex flex-col items-center py-2 px-3 grow\\"><img src=\\"potentiel.svg\\" alt=\\"\\" class=\\"\\">
  <p class=\\"font-dm-sans font-medium text-xs\\" id=\\"step-title\\">Ensoleillement</p>
</div>"
`;
